cloud/new_ec2_client.go

```diff
@@ -78,6 +84,168 @@ func (c *awsClientImpl) Close() {
 	c.httpClient = nil
 }
 
+// RunInstances is a wrapper for ec2.RunInstances.
+func (c *awsClientImpl) RunInstances(input *ec2.RunInstancesInput) (*ec2.Reservation, error) {
+	var output *ec2.Reservation
+	var err error
+	_, err = util.Retry(
+		func() (bool, error) {
+			output, err = c.EC2.RunInstances(input)
+			if err != nil {
+				return true, err
+			}
+			return false, nil
+		}, awsClientImplRetries, awsClientImplStartPeriod)
+	if err != nil {
+		return nil, err
+	}
+	return output, err
+}
+
+// DescribeInstances is a wrapper for ec2.DescribeInstances
+func (c *awsClientImpl) DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	var output *ec2.DescribeInstancesOutput
+	var err error
+	_, err = util.Retry(
+		func() (bool, error) {
+			output, err = c.EC2.DescribeInstances(input)
+			if err != nil {
+				return true, err
+			}
+			return false, nil
+		}, awsClientImplRetries, awsClientImplStartPeriod)
+	if err != nil {
+		return nil, err
+	}
+	return output, err
+}
+
+// CreateTags is a wrapper for ec2.CreateTags.
+func (c *awsClientImpl) CreateTags(input *ec2.CreateTagsInput) (*ec2.CreateTagsOutput, error) {
+	var output *ec2.CreateTagsOutput
+	var err error
+	_, err = util.Retry(
+		func() (bool, error) {
+			output, err = c.EC2.CreateTags(input)
+			if err != nil {
+				return true, err
+			}
+			return false, nil
+		}, awsClientImplRetries, awsClientImplStartPeriod)
+	if err != nil {
+		return nil, err
+	}
+	return output, err
+}
+
+// TerminateInstances is a wrapper for ec2.TerminateInstances.
+func (c *awsClientImpl) TerminateInstances(input *ec2.TerminateInstancesInput) (*ec2.TerminateInstancesOutput, error) {
+	var output *ec2.TerminateInstancesOutput
+	var err error
+	_, err = util.Retry(
+		func() (bool, error) {
+			output, err = c.EC2.TerminateInstances(input)
+			if err != nil {
+				return true, err
+			}
+			return false, nil
+		}, awsClientImplRetries, awsClientImplStartPeriod)
+	if err != nil {
+		return nil, err
+	}
+	return output, err
+}
+
+// RequestSpotInstances is a wrapper for ec2.RequestSpotInstances.
+func (c *awsClientImpl) RequestSpotInstances(input *ec2.RequestSpotInstancesInput) (*ec2.RequestSpotInstancesOutput, error) {
+	var output *ec2.RequestSpotInstancesOutput
+	var err error
+	_, err = util.Retry(
+		func() (bool, error) {
+			output, err = c.EC2.RequestSpotInstances(input)
+			if err != nil {
+				return true, err
+			}
+			return false, nil
+		}, awsClientImplRetries, awsClientImplStartPeriod)
+	if err != nil {
+		return nil, err
+	}
+	return output, err
```

Comment:
the second return value here should be nil, just to be clear.

and we should probably wrap/withstack the actual error case.

Reply:
I fixed the nil. I am not wrapping because there are times when we cast to an awserr and check the message, which would change if we wrapped. I am relying on the caller to wrap.